cmd/encryption-v1.go

```diff
@@ -923,16 +923,18 @@ func getDecryptedETag(headers http.Header, objInfo ObjectInfo, copySource bool)
 	if crypto.IsMultiPart(objInfo.UserDefined) {
 		return objInfo.ETag
 	}
+
+	// If source is SSE-C encrypted and this is a SSE-C Copy operation
+	// we should return the last 32 bytes for the ETag, as there is no
+	// decryption necessary to comply with AWS S3 spec.
 	if crypto.SSECopy.IsRequested(headers) {
-		key, err = crypto.SSECopy.ParseHTTP(headers)
-		if err != nil {
-			return objInfo.ETag
-		}
+		return objInfo.ETag[len(objInfo.ETag)-32:]
```

Comment:
Maybe it's just me but I find the comment quite confusing. In case of SSE-C the etag is encrypted with a separate key (which depends on the object encryption key - see https://github.com/minio/minio/blob/master/cmd/crypto/key.go#L148). Now, the code is *only* correct because the object encryption key is not changed. If the object encryption key would be changed or the etag encryption would depend on e.g. the key-encryption key the could would be wrong b/c the etag would change on a copy... 

Maybe something like:
```
	// If source is SSE-C encrypted and this is a SSE-C Copy operation
	// we should return the last 32 hex chars (16 byte) for the ETag, since the client
        // has only ever seen some this (random-looking) ETag for that object. This corresponds
        // to the S3 spec. since the ETag is not the MD5 sum of the object plaintext in case of SSE-C.
```

Reply:
@poornas does this change work with SSE-C to SSE-S3 - since this returns a different ETag without decrypting how would it work?